stix_shifter_modules/gcp_chronicle/stix_transmission/api_client.py

```diff
@@ -161,12 +164,25 @@ async def delete_search(self, search_id):
 
         delete = self.QUERY_ENDPOINT + "/" + delete_id[1]
         delete_endpoint = self.host + "/" + delete
-        return self.http_client.request(delete_endpoint, 'DELETE')
+        return await self.client_api(delete_endpoint, 'DELETE')
 
     def create_http_client(self):
         """
         Initialize the http client object using the credentials
         :return: None
         """
-        credentials = service_account.Credentials.from_service_account_info(self.auth, scopes=self.QUERY_SCOPES)
-        self.http_client = _auth.authorized_http(credentials)
+        credentials = auth.creds.ServiceAccountCreds(**self.auth, scopes=self.QUERY_SCOPES)
+        self.http_client = Aiogoogle(service_account_creds=credentials)
+
+    async def client_api(self, url, method, data=None):
+        """
+        make an api call using aiogoogle request module
+        :params
+        url(str) -> host url
+        method(str) -> GET, POST
+        data(json) -> query input
+        :return: response
+        """
+        async with self.http_client as client:
+            req = Request(url=url, method=method, data=data)
+            return await client.as_service_account(req, full_res=True)
```

Comment:
we need to use timeout argument as well, when calling `as_service_account`
```
return await client.as_service_account(req, full_res=True, timeout=self.timeout)
``` 

The self.timeout you can get in the class init:
`self.timeout = connection['options'].get('timeout')`

Also, please test with `self.timeout = 1` and handle the timeout error.

Reply:
Thanks for your comments.
As per your suggestion timeout and search id exceptions are handled.